### PR TITLE
CI with `--all-features` and `--no-default-features`

### LINKED ISSUE #399: Modify CI workflow to run with a matrix that checks `--all-features` and `--no-default-features`

### PR DESCRIPTION
**Pull request by pbillaut**
Fix #399.

### CODE REVIEW

**Review by thomaseizinger — COMMENTED** [review]
Thanks!

This is great. I am not sure why the tests are failing though. Have they been broken all this time?

To merge this, we will also need to extend the jobs in `bors.toml`, otherwise bors will listen for jobs that are never going to run.

**Comment by pbillaut** [comment]
> Thanks!
> 
> This is great. I am not sure why the tests are failing though. Have they been broken all this time?
> 
> To merge this, we will also need to extend the jobs in `bors.toml`, otherwise bors will listen for jobs that are never going to run.

Hey, the tests are failing since the `Http` client needs the Docker daemon to run with the Remote API Access enabled. This can for example be achieved by starting the Docker daemon like this:
```bash
dockerd --host fd:// --containerd=/run/containerd/containerd.sock --host tcp://127.0.0.1:2375
```

By default, and thus on the GitHub worker, Remote API Access is not enabled and that's why we're seeing _connection refused_ errors. I'm currently experimenting with running [Docker-in-Docker](https://hub.docker.com/_/docker) to provide the Remote API Access on the worker via the dockerized Docker daemon.

**Comment by pbillaut** [comment]
Instead of running another Docker container that provides the Docker Remote API, it's simpler to just enable it via systemd, of course only exposed on localhost.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,10 +28,25 @@ jobs:
...
+        cargo-cmd:
+          - test --all-features
+          - test --no-default-features
     steps:
+      - name: Enable Docker Remote API on Localhost
+        shell: bash
+        run: |
+          sudo sed -i \
+            's#ExecStart=/usr/bin/dockerd#ExecStart=/usr/bin/dockerd -H tcp://127.0.0.1:2375#g' \
+            /lib/systemd/system/docker.service
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:47]: I guess this is good enough here. We could also consider [installing an override file](https://www.digitalocean.com/community/tutorials/how-to-use-systemctl-to-manage-systemd-services-and-units#editing-unit-files).

What do you think?

Comment (pbillaut) [reply .github/workflows/ci.yml]: I have no real preference here - adapted pipeline accordingly in 5921126.

**Review by thomaseizinger — APPROVED** [review]
Thank you for this and sorry for the delay in getting back to you!

bors r+

**Pull request merged**